pages/quests/index.tsx

```diff
@@ -8,23 +8,113 @@ import {
   Input,
   Stack
 } from '@chakra-ui/react'
-import { useAccount, useConnect } from 'wagmi'
-import { useRouter } from 'next/router'
+import { useEffect, useState } from 'react'
+import { useAccount, useConnect, useNetwork } from 'wagmi'
 import Link from 'next/link'
 import useTranslation from 'next-translate/useTranslation'
 import { Layout } from '@/components/layouts/layout'
+import { useToast } from '@/hooks/useToast'
 import { useMounted } from '@/hooks/useMounted'
 import { useKeywordSubmit } from '@/hooks/quest/useKeywordSubmit'
+import { useUpdateToken } from '@/hooks/quest/useUpdateToken'
 import { useHasNFT } from '@/hooks/useHasNFT'
+import { useTokenIdOf } from '@/hooks/useTokenIdOf'
+import { useTokenURI } from '@/hooks/useTokenURI'
+import { getContractAddress } from '@/utils/contractAddress'
+import { useUpdateTokenMetadata, KamonToken } from '@/hooks/useUpdateTokenMetadata'
 
 const Quests: NextPage = () => {
   const { t } = useTranslation('common')
+  const { toast } = useToast()
   const mounted = useMounted()
   const { connect, connectors } = useConnect()
   const [metaMask] = connectors
   const { data } = useAccount()
-  const { keyword, inputChange, submit, isSubmitting } = useKeywordSubmit()
+  const { keyword, inputChange, submit, isSubmitting, keywordSubmitSucceeded } = useKeywordSubmit()
   const { hasNFT } = useHasNFT()
+  const { activeChain } = useNetwork()
+  const kamonNFT = getContractAddress({
+    name: 'kamonNFT',
+    chainId: activeChain?.id
+  })
+  const { tokenIdOf } = useTokenIdOf(kamonNFT, data?.address)
+  const { tokenURI } = useTokenURI(kamonNFT, tokenIdOf?.toNumber() || 0)
+  const [tokenId, setTokenId] = useState<BigInt>(BigInt(0))
+  const [tokenJSON, setTokenJSON] = useState<KamonToken>()
+  const [finalTokenUri, setFinalTokenUri] = useState('')
+  const [updateTxLaunched, setUpdateTxLaunched] = useState<boolean>()
+
+  const { updateTokenMetadata, updateTokenMetadataIsSubmitting } = useUpdateTokenMetadata()
+  const { updateToken, updateTokenIsSubmitting } = useUpdateToken(
+    kamonNFT,
+    tokenId,
+    finalTokenUri,
+  )
+
+  useEffect(() => {
+    if (tokenURI) {
+      const fetchData = async () => {
+        const pinataRequest = await fetch(tokenURI.toString())
+        const responseJson = await pinataRequest.json()
+        setTokenJSON(responseJson)
+      }
+      fetchData()
+    }
+  }, [tokenURI])
+
+  const updateTokenMetadataWrapper = async () => {
```

Comment:
can you separate to different files?
cuz this quest/index is quit big already

Reply:
@SamuraiT, I thought we wanted to put the toast calls on this file.

Does this change after the localization system change?

Reply:
better to be separated. cuz this quests/index becoming too big.

Reply:
This is still pending, sorry.

Considering moving the code to the `useUpdateTokenMetadata.ts` hook file. Hoping that I will be able to localize the toasts.

Reply:
Done in https://github.com/henkaku-center/omise-interface/pull/80/commits/3e55d7b61d5b73ede2882d972777fd220343652a